.github/workflows/pr-build.yml

```diff
@@ -81,6 +82,10 @@ jobs:
         with:
           arguments: testing:validator:build
 
+      - name: Dry Run Validator
```

Comment:
Is the idea to catch compilation failures as well, also can you please check if it needs to run all os [reference here](https://github.com/aws-observability/aws-otel-java-instrumentation/pull/738/files#diff-404033996c34f9971e0c112d8538b9063618fc85059a44df7d4db448974888caR81).